src/bikeracks/votes.py

```diff
@@ -69,17 +69,26 @@ def submit_vote():
     # connect to db
     db = get_db()
     
+    # check if user has voted on this rack before
+    vote_status = h.get_vote_status(db, rack_id, user_id)
+    
     query = """ INSERT INTO votes (rack_id, user_id, vote_type)
                 VALUES (?, ?, ?)
                 ON CONFLICT(rack_id, user_id) DO UPDATE SET vote_type=?"""
-    db.execute(query, (rack_id, user_id, vote_type))
+    db.execute(query, (rack_id, user_id, vote_type, vote_type))
     db.commit()
     
-    # if the vote is an upvote then delta_up is 1 and delta_down is -1
-    if vote_type == 1:
+    # if the vote is an upvote and the user has voted before
+    # then delta_up is 1 and delta_down is -1
+    
+    if vote_type == 1 and voteStatus:
```

Comment:
what is `voteStatus`? An int representing the old vote value? A bool representing whether or not there is a previous vote of any kind?

Reply:
Clarified in a comment what it is.